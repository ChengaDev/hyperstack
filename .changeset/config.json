{
  "$schema": "https://unpkg.com/@changesets/config@2.0.0/schema.json",
  "changelog": false,
  "commit": false,
  "linked": [],
  "access": "public",
  "baseBranch": "master",
  "updateInternalDependencies": "patch",
  "ignore": [],
  "fixed": [
    [
      "create-hyperstack",
      "template-app",
      "template-blank",
      "hyperstack",
      "@hyperstackjs/*"
    ]
  ]
}
