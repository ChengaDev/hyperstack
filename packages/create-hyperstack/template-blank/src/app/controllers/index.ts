// naming and imports: name and import as you see fit

const controllers: any[] = []
// controllers.push(NotesController)

export const AppController = {
  controllers,
}
