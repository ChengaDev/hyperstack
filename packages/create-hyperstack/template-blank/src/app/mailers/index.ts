// suggested mailer layout:
// mailer-name.ts
// mailer-name/
//  template-contents
//
// export your mailers here:
// export { MyMailer } from './my-mailer'

export {}
