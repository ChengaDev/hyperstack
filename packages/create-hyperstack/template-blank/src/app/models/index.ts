// export your models here
// export { MyModel } from './my-model'

export {}
