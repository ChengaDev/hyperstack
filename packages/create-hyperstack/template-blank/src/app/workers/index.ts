// export your workers here:
// e.g. export { MyWorker } from './my-worker'
export {}
