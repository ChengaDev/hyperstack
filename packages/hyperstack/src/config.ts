import path from 'path'
import L from 'lodash'
import createDebug from 'debug'
import type { Config } from '@hyperstackjs/typings'
import { getAppEnvironment } from './context'
import { requireModule } from './utils'
import type { ConfigPaths, Context } from './types'
const debug = createDebug('hyperstack:config')

export const resolveConfig = ({ exactPath, appPath }: ConfigPaths) => {
  if (exactPath) {
    return exactPath
  }
  if (!appPath) {
    throw new Error('app path is empty')
  }
  return path.join(appPath, `config/environments/${getAppEnvironment()}`)
}
export const loadConfig = async (paths: ConfigPaths) => {
  const resolved = resolveConfig(paths)

  const mod = requireModule(resolved)

  // configs are function providing a dict.
  // it's a function because merging modifies the provided dict, and we
  // don't want to fiddle with module cache in case we want live reloads or tests to work
  const config = (await mod()) as Config

  debug('config before overrides is %o', config)
  if (paths.configOverrides) {
    L.merge(config, paths.configOverrides)
  }
  return { config, resolvedConfigPath: resolved }
}

export const tieConfig = async (
  context: Context,
  { exactPath, configOverrides }: ConfigPaths = {}
) => {
  const appPath = context.store().app!.root
  const { config, resolvedConfigPath } = await loadConfig({
    configOverrides,
    exactPath,
    appPath,
  })
  debug('loaded from %o', resolvedConfigPath)
  debug('config is %o', config)
  debug('overrides %o', configOverrides)
  context.setConfig(config)
}
