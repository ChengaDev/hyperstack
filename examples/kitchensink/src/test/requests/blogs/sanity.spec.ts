import { test } from '@hyperstackjs/testing'
import { root } from '../../../config/settings'
import { appContext } from '../../../app'

const {
  requests,
  matchers: { matchRequestWithSnapshot },
} = test(root)

describe('requests', () => {
  describe('/blogs', () => {
    requests('all', async (request) => {
      const { Blog } = appContext.models()
      await Blog.create({
        content: 'interesting blog',
        title: 'this is the title',
      })

      await matchRequestWithSnapshot(200, request().get(`/blogs`))
    })
    requests('create', async (request) => {
      await matchRequestWithSnapshot(
        200,
        request().post(`/blogs`).send({
          content: 'interesting blog',
          title: 'this is the title',
        })
      )
    })
    requests('get :id', async (request) => {
      const { Blog } = appContext.models()
      const blog = await Blog.create({
        content: 'interesting blog',
        title: 'this is the title',
      })

      await matchRequestWithSnapshot(200, request().get(`/blogs/${blog.id}`))
    })
    requests('update :id', async (request) => {
      const { Blog } = appContext.models()
      const blog = await Blog.create({
        content: 'interesting blog',
        title: 'this is the title',
      })
      await request()
        .post(`/blogs/${blog.id}`)
        .send({
          content: 'interesting blog' + 'update',
          title: 'this is the title' + 'update',
        })

      await matchRequestWithSnapshot(200, request().get(`/blogs/${blog.id}`))
    })
  })
})
