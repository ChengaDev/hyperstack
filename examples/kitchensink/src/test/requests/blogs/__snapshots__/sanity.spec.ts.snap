// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requests /blogs all 1`] = `
{
  "body": {
    "blogs": [
      {
        "content": "interesting blog",
        "title": "this is the title",
      },
    ],
  },
  "headers": {
    "connection": "close",
    "content-length": "70",
    "content-security-policy": "default-src 'self';base-uri 'self';block-all-mixed-content;font-src 'self' https: data:;form-action 'self';frame-ancestors 'self';img-src 'self' data:;object-src 'none';script-src 'self';script-src-attr 'none';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests",
    "content-type": "application/json; charset=utf-8",
    "cross-origin-embedder-policy": "require-corp",
    "cross-origin-opener-policy": "same-origin",
    "cross-origin-resource-policy": "same-origin",
    "expect-ct": "max-age=0",
    "origin-agent-cluster": "?1",
    "referrer-policy": "no-referrer",
    "strict-transport-security": "max-age=15552000; includeSubDomains",
    "vary": "Accept-Encoding",
    "x-content-type-options": "nosniff",
    "x-dns-prefetch-control": "off",
    "x-download-options": "noopen",
    "x-frame-options": "SAMEORIGIN",
    "x-permitted-cross-domain-policies": "none",
    "x-xss-protection": "0",
  },
  "status": 200,
}
`;

exports[`requests /blogs create 1`] = `
{
  "body": {
    "blog": {
      "content": "interesting blog",
      "title": "this is the title",
    },
  },
  "headers": {
    "connection": "close",
    "content-length": "67",
    "content-security-policy": "default-src 'self';base-uri 'self';block-all-mixed-content;font-src 'self' https: data:;form-action 'self';frame-ancestors 'self';img-src 'self' data:;object-src 'none';script-src 'self';script-src-attr 'none';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests",
    "content-type": "application/json; charset=utf-8",
    "cross-origin-embedder-policy": "require-corp",
    "cross-origin-opener-policy": "same-origin",
    "cross-origin-resource-policy": "same-origin",
    "expect-ct": "max-age=0",
    "origin-agent-cluster": "?1",
    "referrer-policy": "no-referrer",
    "strict-transport-security": "max-age=15552000; includeSubDomains",
    "vary": "Accept-Encoding",
    "x-content-type-options": "nosniff",
    "x-dns-prefetch-control": "off",
    "x-download-options": "noopen",
    "x-frame-options": "SAMEORIGIN",
    "x-permitted-cross-domain-policies": "none",
    "x-xss-protection": "0",
  },
  "status": 200,
}
`;

exports[`requests /blogs get :id 1`] = `
{
  "body": {
    "blog": {
      "content": "interesting blog",
      "title": "this is the title",
    },
  },
  "headers": {
    "connection": "close",
    "content-length": "67",
    "content-security-policy": "default-src 'self';base-uri 'self';block-all-mixed-content;font-src 'self' https: data:;form-action 'self';frame-ancestors 'self';img-src 'self' data:;object-src 'none';script-src 'self';script-src-attr 'none';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests",
    "content-type": "application/json; charset=utf-8",
    "cross-origin-embedder-policy": "require-corp",
    "cross-origin-opener-policy": "same-origin",
    "cross-origin-resource-policy": "same-origin",
    "expect-ct": "max-age=0",
    "origin-agent-cluster": "?1",
    "referrer-policy": "no-referrer",
    "strict-transport-security": "max-age=15552000; includeSubDomains",
    "vary": "Accept-Encoding",
    "x-content-type-options": "nosniff",
    "x-dns-prefetch-control": "off",
    "x-download-options": "noopen",
    "x-frame-options": "SAMEORIGIN",
    "x-permitted-cross-domain-policies": "none",
    "x-xss-protection": "0",
  },
  "status": 200,
}
`;

exports[`requests /blogs update :id 1`] = `
{
  "body": {
    "blog": {
      "content": "interesting blogupdate",
      "title": "this is the titleupdate",
    },
  },
  "headers": {
    "connection": "close",
    "content-length": "79",
    "content-security-policy": "default-src 'self';base-uri 'self';block-all-mixed-content;font-src 'self' https: data:;form-action 'self';frame-ancestors 'self';img-src 'self' data:;object-src 'none';script-src 'self';script-src-attr 'none';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests",
    "content-type": "application/json; charset=utf-8",
    "cross-origin-embedder-policy": "require-corp",
    "cross-origin-opener-policy": "same-origin",
    "cross-origin-resource-policy": "same-origin",
    "expect-ct": "max-age=0",
    "origin-agent-cluster": "?1",
    "referrer-policy": "no-referrer",
    "strict-transport-security": "max-age=15552000; includeSubDomains",
    "vary": "Accept-Encoding",
    "x-content-type-options": "nosniff",
    "x-dns-prefetch-control": "off",
    "x-download-options": "noopen",
    "x-frame-options": "SAMEORIGIN",
    "x-permitted-cross-domain-policies": "none",
    "x-xss-protection": "0",
  },
  "status": 200,
}
`;
